src/mlpack/methods/reinforcement_learning/q_networks/vanilla_dqn.hpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file vanilla_dqn.hpp
+ * @author Nishant Kumar
+ *
+ * This file contains the implementation of the vanilla deep q network.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_VANILLA_DQN_HPP
+#define MLPACK_METHODS_RL_VANILLA_DQN_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/init_rules/gaussian_init.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/loss_functions/mean_squared_error.hpp>
+
+namespace mlpack {
+namespace rl {
+
+using namespace mlpack::ann;
+
+template <typename NetworkType = FFN<MeanSquaredError<>,
+                                    GaussianInitialization>>
+class VanillaDQN
```

Comment:
Do you mind to go with another name, I think most people will associate the name with the network from the paper, which isn't correct. Maybe `SimpleDQN` or `MinimalDQN` work as well, maybe you have another name in mind?

Reply:
Yeah, I guess `SimpleDQN` should do.. Sounds more relatable